lib/Conversion/LTLToCore/LTLToCore.cpp

```diff
@@ -0,0 +1,237 @@
+//===- LTLToCore.cpp -----------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// Converts LTL and Verif operations to Core operations
+//
+//===----------------------------------------------------------------------===//
+
+#include "circt/Conversion/LTLToCore.h"
+#include "../PassDetail.h"
+#include "circt/Dialect/Comb/CombOps.h"
+#include "circt/Dialect/HW/HWOps.h"
+#include "circt/Dialect/LTL/LTLDialect.h"
+#include "circt/Dialect/LTL/LTLOps.h"
+#include "circt/Dialect/SV/SVDialect.h"
+#include "circt/Dialect/SV/SVOps.h"
+#include "circt/Dialect/Seq/SeqOps.h"
+#include "circt/Dialect/Verif/VerifOps.h"
+#include "circt/Support/BackedgeBuilder.h"
+#include "circt/Support/Namespace.h"
+#include "mlir/Dialect/Func/IR/FuncOps.h"
+#include "mlir/Dialect/LLVMIR/LLVMDialect.h"
+#include "mlir/Dialect/SCF/IR/SCF.h"
+#include "mlir/Transforms/DialectConversion.h"
+
+using namespace mlir;
+using namespace circt;
+using namespace hw;
+
+//===----------------------------------------------------------------------===//
+// Conversion patterns
+//===----------------------------------------------------------------------===//
+
+namespace {
+/// Lower a ltl::DisableOp operation to Core operations
+struct DisableOpConversion : OpConversionPattern<ltl::DisableOp> {
+  using OpConversionPattern<ltl::DisableOp>::OpConversionPattern;
+
+  // DisableOp translates to an implication in the form of
+  // ~condition -> input
+  LogicalResult
+  matchAndRewrite(ltl::DisableOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+
+    // Replace the ltl::DisableOp with an OR op as it represents a disabling
+    // implication: (implies (not condition) input) is equivalent to
+    // (or (not (not condition)) input) which becomes (or condition input)
+    rewriter.replaceOpWithNewOp<comb::OrOp>(op, adaptor.getCondition(),
+                                            adaptor.getInput());
+
+    return success();
+  }
+};
+
+struct HasBeenResetOpConversion : OpConversionPattern<verif::HasBeenResetOp> {
+  using OpConversionPattern<verif::HasBeenResetOp>::OpConversionPattern;
+
+  // HasBeenReset generates a 1 bit register that is set to one if the reset
+  // value is found to be active at any point
```

Comment:
Looking through the LTL to SV lowering, I noticed that they don't actually do this, rather they make the disable signal depend on `(not (and hbr (not reset)))` rather than simply `hbr`. I think that it's probably easier to implement that way and the final behavior is the same as has been reset should only be used in a disable condition normally.